etl/steps/data/meadow/hmd/2022-12-07/life_tables.py

```diff
@@ -0,0 +1,400 @@
+    # df to table
+    table = df_to_table(df, age, year, folder["sex"])
+    # # underscore all table columns
+    table = underscore_table(table)
+    return table
+
+
+def _age_year(input_folder: str) -> Tuple[str, str]:
+    ageyear = input_folder.split("_")[-1]
+    age, year = ageyear.split("x")
+    return age, year
+
+
+def make_df(files: List[str], regex_header: str) -> pd.DataFrame:
+    """Create dataframe.
+
+    Parameters
+    ----------
+    files : List[str]
+        Files to load and extract data from. There is a file per country. Within the file, data is found
+        along with country name. Note that sometimes, the country name can be like "New Zealand - Non-Maori".
+    regex_header: str
+        Regex of the file header text line.
+
+    Returns
+    -------
+    pd.DataFrame
+        _description_
+    """
+    log.info("hmd_lt: creating dataframe from files...")
+    dfs = []
+    # Load each file
+    for f in files:
+        with open(f, "r") as f:
+            text = f.read()
+        # Get relevant fields
+        match = re.search(regex_header, text)
+        if match is not None:
+            country, _, table_str = match.group(1, 2, 3)
+        else:
+            raise ValueError(f"No match found! Please revise that source files' content matches {regex_header}.")
+        # Build country df
+        df_ = _make_df_country(country, table_str)
+        dfs.append(df_)
+    # Concatenate all country dfs
+    df = pd.concat(dfs, ignore_index=True)
+    return cast(pd.DataFrame, df)
+
+
+def _make_df_country(country: str, table: str) -> pd.DataFrame:
+    """Create dataframe for individual country."""
+    # Remove starting/ending spaces
+    table = table.strip()
+    # Remove spacing after newline
+    table = re.sub(r"\n\s+", "\n", table)
+    # Replace spacing with tabs
+    table = re.sub(r"[^\S\r\n]+", "\t", table)
+    # Build df
+    df = pd.read_csv(StringIO(table), sep="\t")
+    # Assign country
+    df = df.assign(Country=country)
+    # # Filter columns
+    # df = df[["Country", "Year", "ex"]].rename(columns={"ex": "life_expectancy"})
+    return df
+
+
+def clean_df(df: pd.DataFrame) -> pd.DataFrame:
+    """Clean dataframe.
+
+    Orders columns, renames columns, checks for missing values and sets dtypes.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        Initial dataframe.
+
+    Returns
+    -------
+    pd.DataFrame
+        Cleaned dataframe.
+    """
+    log.info("hmd_lt: cleaning dataframe...")
+    # Order columns
+    cols_first = ["Country", "Year"]
+    cols = cols_first + [col for col in df.columns if col not in cols_first]
+    df = df[cols]
+    # Rename columns
+    df = _clean_rename_columns_df(df)
+    # Correct missing data
+    df = _clean_correct_missing_data(df)
+    # Set dtypes
+    df = _clean_set_dtypes_df(df)
+    return df
+
+
+def _clean_rename_columns_df(df: pd.DataFrame) -> pd.DataFrame:
+    """Rename columns."""
+    return df.rename(columns=COLUMNS_RENAME)
```

Comment:
An additional check here would be to add `errors='raise'` to ensure that all columns to be renamed are found.